test/functional/api/rpc_spec.lua

```diff
@@ -0,0 +1,47 @@
+local helpers = require('test.functional.helpers')(after_each)
```

Comment:
The only thing "unusual" is possibly the split between files on requests vs notification does not fit all tests, it is possible it can be better organized. But arbitrarily adding some new tests in a third file will not help.

 "Manually controlling the event loop" is not "unusual approach" and nothing you need to split a file on, it is something one _just does_ in a test exactly whenever the test needs it.